From: Manuel Bachmann <manuel.bachmann@open.eurogiciel.org>
Date: Thu, 12 Jun 2014 15:51:27 +0200
Subject: downstream: accept the new "--enable-sys-uid" configure option

This configure option activates the code allowing non-root users
(such as "display") to launch Weston directly.

Change-Id: I844ce7aaba717d1453092e62885b704e9473c4f7
Signed-off-by: Manuel Bachmann <manuel.bachmann@open.eurogiciel.org>
---
 Makefile.am  | 4 ++++
 configure.ac | 7 +++++++
 2 files changed, 11 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 343adc6..239a27a 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -33,6 +33,10 @@ AM_CPPFLAGS = 					\
 	-DLIBEXECDIR='"$(libexecdir)"'		\
 	-DBINDIR='"$(bindir)"'
 
+if ENABLE_SYS_UID
+AM_CPPFLAGS += -DLOCAL_CONFIG_WESTON_ALLOW_SYS_UID=1
+endif
+
 CLEANFILES = weston.ini $(BUILT_SOURCES)
 
 bin_PROGRAMS += weston
diff --git a/configure.ac b/configure.ac
index f55cea8..d7bb414 100644
--- a/configure.ac
+++ b/configure.ac
@@ -69,6 +69,13 @@ if test x$enable_egl = xyes; then
         PKG_CHECK_MODULES([EGL_TESTS], [egl >= 7.10 glesv2 wayland-client wayland-egl])
 fi
 
+AC_ARG_ENABLE(sys_uid, [  --enable-sys-uid],,
+              enable_sys_uid=no)
+AM_CONDITIONAL(ENABLE_SYS_UID, test x$enable_sys_uid = xyes)
+if test x$enable_sys_uid = xyes; then
+	AC_DEFINE([ENABLE_SYS_UID], [1], [Allow system users to launch Weston without weston-launch])
+fi
+
 AC_ARG_ENABLE(xkbcommon,
 	      AS_HELP_STRING([--disable-xkbcommon], [Disable libxkbcommon
 			      support: This is only useful in environments
